### ISSUE TITLE
Search in the tree causes a crash

### ISSUE THREAD

**Issue by Futhark1988** [giuspen/cherrytree#1974]
**Version, Operative system**
Windows 10 Enterprise, 20H2

**Describe the bug**
(Repeated) Search in the tree causes CherryTree to crash.

**To Reproduce**
1. Click on any node in the tree on the left hand-side.
2. Start typing the name of a node as fast as you can, and you will see that the courser doesn't get placed at the end of the input in time. Which results in gibberish input.
3. CherryTree closes/crashes.

**Comment by princessdaphie** [comment]
I have the same issue, in 2025, Windows 10 Pro 22H2, cherrytree v1.4.0. If I use the search by typing feature in the tree, it often crashes.

**Comment by daPhie79** [comment]
I have the same issue, in 2025, Windows 10 Pro 22H2, cherrytree v1.4.0. If I use the search by typing feature in the tree, it often crashes.

**Comment by giuspen** [comment]
@daPhie79 could you please test either https://www.giuspen.net/software/cherrytree_1.4.0.0+64_win64_portable.7z or https://www.giuspen.net/software/cherrytree_1.4.0.0+64_win64_setup.exe and report if you can still reproduce the crash?

**Comment by daPhie79** [comment]
> [@daPhie79](https://github.com/daPhie79) could you please test either https://www.giuspen.net/software/cherrytree_1.4.0.0+64_win64_portable.7z or https://www.giuspen.net/software/cherrytree_1.4.0.0+64_win64_setup.exe and report if you can still reproduce the crash?

Thanks for working on this issue!

I tried your dev build and it still does it. If it helps, what seems to happen is that when I type too fast, it crashes, and the way it crashes looks like to be an infinite loop. If I go very slow between each letter, it doesn't crash.

**Comment by daPhie79** [comment]
Alright, I tested your new build. I'm both happy to see that typing fast doesn't make it crash anymore, but a bit sad that you've completely changed the quick search function, I found it more intuitive before with the simple floating box and the live updating tree position. Is it because this procedure is impossible to fix that you've overhauled it? Although I gotta say the new search is also pretty impressive.

**Comment by giuspen** [comment]
Yes @daPhie79 I did prefer to merge the two searches because the gtk tree view embedded library search is not easy for me to control while the so called quick search, which already existed but was launched with the Ctrl+Shift+G, is under control.
The embedded search also doesn't auto expand a hidden/collapsed node and works only if you type from the first character, not part of the name.
Still if you don't digest the new search let me know and I will add an option to use the old/embedded one, trying to sort out the crash.

**Comment by daPhie79** [comment]
It would be cool to be able to switch between both, haha, but I certainly could get used to the new one, it's pretty nifty!
